__tests__/events.ts

```diff
@@ -0,0 +1,167 @@
+import globalEvents  from "../src/utils/events";
+import { EventTargetOptions } from "../src/types/events"
+import useTask, { Task } from "../src/Task";
+import { mockSetup } from "../test-utils/components";
+import { TaskInstance } from "../src/TaskInstance";
+
+
+describe("events", () => {
+    it("adds event handler", async () => {
+      const handler = () => {}
+      const {addEvent, hasEvent} = globalEvents();
+      const eventHandler = {
+        target: EventTargetOptions.OnError,
+        key: "test",
+        handler
+      };
+  
+      addEvent(eventHandler);
+  
+      expect(hasEvent(eventHandler)).toBe(true);
+    });
```

Comment:
These tests look good but I'd prefer if they'd test the final thing (public API). Let's see if they can be adjusted later?

Reply:
Oh agreed, I just tested what I had so far.

Reply:
Added `events-api` for testing the public api use of it